phoenix-core/src/it/java/org/apache/phoenix/end2end/index/ImmutableIndexIT.java

```diff
@@ -259,7 +276,136 @@ private void assertIndexMutations(Connection conn) throws SQLException {
                 (transactionProvider != null && 
                  transactionProvider.isUnsupported(Feature.MAINTAIN_LOCAL_INDEX_ON_SERVER)), iterator.hasNext());
     }
-    
+
+    private void createAndPopulateTableAndIndexForConsistentIndex(Connection conn, String tableName, String indexName, int numOfRowsToInsert)
+            throws Exception {
+        String ddl = "CREATE TABLE " + TABLE_NAME + TestUtil.TEST_TABLE_SCHEMA + tableDDLOptions;
+        INDEX_DDL =
+                "CREATE " + " INDEX IF NOT EXISTS " + SchemaUtil.getTableNameFromFullName(indexName)
+                        + " ON " + tableName + " (long_pk, varchar_pk)"
+                        + " INCLUDE (long_col1, long_col2) ";
+
+        conn.createStatement().execute(ddl);
+        conn.createStatement().execute(INDEX_DDL);
+        upsertRows(conn, tableName, numOfRowsToInsert);
+        conn.commit();
+
+        TestUtil.waitForIndexState(conn, indexName, PIndexState.ACTIVE);
+    }
+
+    @Test
+    public void testGlobalImmutableIndexCreate() throws Exception {
+        if (localIndex || transactionProvider != null) {
+            return;
+        }
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = "TBL_" + generateUniqueName();
+        String indexName = "IND_" + generateUniqueName();
+        String fullTableName = SchemaUtil.getTableName(TestUtil.DEFAULT_SCHEMA_NAME, tableName);
+        String fullIndexName = SchemaUtil.getTableName(TestUtil.DEFAULT_SCHEMA_NAME, indexName);
+        TABLE_NAME = fullTableName;
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            int numRows = 1;
+            createAndPopulateTableAndIndexForConsistentIndex(conn, fullTableName, fullIndexName,
+                    numRows);
+
+            ResultSet rs;
+            rs = conn.createStatement().executeQuery("SELECT /*+ NO_INDEX */ COUNT(*) FROM " + TABLE_NAME);
+            assertTrue(rs.next());
+            assertEquals(numRows, rs.getInt(1));
+            rs = conn.createStatement().executeQuery("SELECT COUNT(*) FROM " + fullIndexName);
+            assertTrue(rs.next());
+            assertEquals(numRows, rs.getInt(1));
+            verifyRowsForEmptyColValue(conn, fullIndexName, IndexRegionObserver.VERIFIED_BYTES);
```

Comment:
The return value of verifyRowsForEmptyColValue() is not checked here and in the other places. Is it intentional? If so, what is the purpose of calling verifyRowsForEmptyColValue().

Reply:
good catch!